# Ultralytics Framework for YOLOv10 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolov08s Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolov8s.pt")

# Validate the pretrained weights by running inference on an image.

In [4]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 15.2ms
Speed: 20.7ms preprocess, 15.2ms inference, 2.4ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [10]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [11]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

New https://pypi.org/project/ultralytics/8.3.13 available 😃 Update with 'pip install -U ultralytics'


Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
engine/trainer: task=detect, mode=train, model=yolov8s.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s, name=train5, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=True, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True,

train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.001), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      7.51G      1.141      2.572      1.109          5        640: 100%|██████████| 36/36 [00:10<00:00,  3.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.60it/s]


                   all       1274       7377       0.73      0.415      0.464      0.298

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100       7.3G     0.9071     0.7717      0.961         17        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.51it/s]


                   all       1274       7377       0.68      0.417       0.45       0.25

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      7.42G      0.917     0.7318     0.9788         37        640: 100%|██████████| 36/36 [00:07<00:00,  4.50it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.591       0.35      0.344      0.212

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      7.44G     0.9185     0.7211     0.9839          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.716      0.454      0.501      0.294

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      7.44G     0.8966     0.7008     0.9681          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.752      0.454      0.509      0.311

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      7.43G     0.8639     0.6351     0.9472         26        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.55it/s]


                   all       1274       7377      0.777      0.488       0.57      0.349

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100       7.3G     0.8794     0.6194     0.9623          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.796      0.491      0.551      0.336

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      7.45G     0.8509      0.593     0.9342         24        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.47it/s]


                   all       1274       7377      0.801      0.509      0.563      0.342

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      7.36G     0.8143     0.5547     0.9407          1        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.777       0.52       0.58       0.36

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      7.42G      0.827     0.5526     0.9261          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.55it/s]


                   all       1274       7377      0.795      0.547      0.607      0.376

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      7.43G     0.8384     0.5538     0.9216         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.794      0.555      0.611      0.371

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      7.45G     0.8398     0.5531     0.9116          1        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.795       0.54      0.599      0.365

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      7.29G     0.8137      0.535     0.9237         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.45it/s]


                   all       1274       7377      0.802      0.536      0.607      0.372

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      7.42G     0.8141     0.5531     0.9189         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.45it/s]


                   all       1274       7377      0.818      0.548      0.628      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      7.46G     0.7914     0.5115     0.9117         24        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.58it/s]


                   all       1274       7377      0.831      0.577       0.63      0.388

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      7.28G     0.7853      0.516     0.9184          4        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.58it/s]


                   all       1274       7377      0.801      0.574       0.63      0.389

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      7.29G     0.7951     0.5226      0.924         15        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.58it/s]


                   all       1274       7377       0.81      0.572      0.638      0.393

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      7.36G     0.7849     0.4994       0.94          6        640: 100%|██████████| 36/36 [00:07<00:00,  4.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.56it/s]


                   all       1274       7377      0.827      0.578      0.641      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      7.52G     0.7881     0.4873     0.9149         12        640: 100%|██████████| 36/36 [00:08<00:00,  4.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.55it/s]


                   all       1274       7377      0.844      0.561      0.638        0.4

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      7.26G      0.767     0.4877     0.9109         23        640: 100%|██████████| 36/36 [00:08<00:00,  4.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.843      0.582      0.656      0.412

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      7.28G     0.7505      0.474     0.9057          4        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.55it/s]


                   all       1274       7377      0.838      0.583      0.653      0.412

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      7.26G     0.7786      0.491     0.9093          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.47it/s]


                   all       1274       7377      0.842      0.588      0.658      0.409

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      7.32G     0.7281     0.4671     0.9005         27        640: 100%|██████████| 36/36 [00:08<00:00,  4.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.857      0.565      0.646      0.409

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      7.45G      0.734     0.4605     0.8995          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.858      0.583      0.668      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      7.36G     0.7388     0.4723     0.9018         20        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.841      0.597      0.665      0.425

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      7.29G     0.7215     0.4569     0.9082         18        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.835      0.597       0.67      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      7.28G     0.7506      0.473     0.8976         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.832      0.601      0.661      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      7.42G     0.7244     0.4542     0.8959         20        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.827      0.605      0.666      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      7.27G     0.6929     0.4335     0.8855          7        640: 100%|██████████| 36/36 [00:08<00:00,  4.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.89it/s]


                   all       1274       7377      0.857      0.602      0.679      0.431

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      7.44G     0.7314     0.4447     0.8993         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.49it/s]


                   all       1274       7377      0.863      0.602      0.678      0.424

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      7.35G     0.7015     0.4377     0.8925         19        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.873      0.594      0.678      0.433

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      7.26G     0.6932     0.4365     0.8822         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.51it/s]


                   all       1274       7377      0.843      0.603       0.68      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      7.45G     0.6954     0.4325     0.8888         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.854        0.6      0.672      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      7.27G     0.7025     0.4291     0.8934         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.846      0.606      0.671      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      7.26G     0.7164     0.4321     0.8974         21        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.829      0.616      0.674      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      7.27G      0.692     0.4242     0.8887          9        640: 100%|██████████| 36/36 [00:08<00:00,  4.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377       0.85       0.61      0.682      0.441

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      7.41G      0.696      0.427     0.8905         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.56it/s]


                   all       1274       7377      0.857       0.62      0.692      0.445

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      7.41G     0.6647     0.5761     0.8704          0        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377      0.864       0.61      0.689      0.444

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      7.36G     0.6791     0.4117     0.8867         21        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.828      0.619      0.682      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      7.43G     0.6796     0.4132     0.8841         15        640: 100%|██████████| 36/36 [00:08<00:00,  4.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.873      0.612      0.694      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      7.26G     0.6697     0.4147     0.8869         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.841      0.622      0.688      0.441

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      7.26G     0.6737     0.4056      0.885         24        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.49it/s]


                   all       1274       7377      0.859      0.612      0.694      0.448

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      7.24G     0.6556     0.4093     0.8904          4        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.857      0.606       0.69      0.451

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      7.44G     0.6839     0.4078     0.8726         26        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377       0.86      0.618        0.7       0.45

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      7.24G     0.6594     0.3998     0.8835          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.859      0.624      0.694       0.45

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      7.27G     0.6457     0.3925     0.8791          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.832      0.632      0.699      0.451

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      7.36G     0.6531     0.3875     0.8744         15        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.849      0.626      0.699      0.456

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      7.41G     0.6687     0.3902     0.8787         16        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.44it/s]


                   all       1274       7377      0.857      0.632      0.701      0.455

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      7.36G     0.6875     0.4643     0.8638          7        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.00it/s]


                   all       1274       7377      0.885      0.618      0.701      0.456

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      7.43G     0.6641     0.3864     0.8752         17        640: 100%|██████████| 36/36 [00:08<00:00,  4.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.851      0.626      0.704      0.461

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      7.28G     0.6549     0.3863     0.8861         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.46it/s]


                   all       1274       7377      0.826      0.632        0.7      0.458

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      7.26G     0.6493       0.38     0.8815          5        640: 100%|██████████| 36/36 [00:08<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.89it/s]


                   all       1274       7377       0.87      0.614      0.709      0.464

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      7.36G     0.6372     0.3828     0.8784          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.87it/s]


                   all       1274       7377      0.868      0.622      0.704      0.452

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      7.42G     0.6227     0.3705     0.8725         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.40it/s]


                   all       1274       7377      0.867       0.62      0.706      0.462

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      7.27G     0.6478     0.3939     0.8767         21        640: 100%|██████████| 36/36 [00:08<00:00,  4.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.864      0.629       0.71      0.466

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      7.35G     0.6207     0.3794     0.8717          5        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377       0.86      0.623      0.707      0.463

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      7.29G     0.6389     0.3808      0.868         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.851      0.632      0.706      0.463

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      7.36G     0.6254     0.3728      0.875         21        640: 100%|██████████| 36/36 [00:08<00:00,  4.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377       0.87      0.621      0.708      0.467

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      7.43G     0.6367     0.3688     0.8691         12        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377      0.867      0.625      0.701      0.463

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      7.41G      0.624     0.3651     0.8697         18        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.62it/s]


                   all       1274       7377      0.881       0.62      0.716      0.475

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      7.44G      0.637     0.3697     0.8693         39        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.65it/s]


                   all       1274       7377      0.875      0.629      0.713       0.47

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      7.44G     0.6119     0.3617     0.8681         33        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.51it/s]


                   all       1274       7377      0.889      0.619      0.712      0.468

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      7.42G     0.6306     0.3687     0.8684         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.852      0.638       0.72      0.477

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      7.35G     0.6423     0.3682      0.866          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.51it/s]


                   all       1274       7377      0.857       0.64      0.714      0.473

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      7.27G     0.6439     0.3659      0.869          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.50it/s]


                   all       1274       7377      0.862      0.638      0.722      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      7.44G     0.6302     0.3629      0.876         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377       0.85      0.648      0.714      0.472

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      7.28G      0.625     0.3644     0.8666         19        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.864      0.638      0.719      0.475

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      7.35G     0.6038     0.3508     0.8572         19        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.878      0.634      0.718      0.476

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      7.29G     0.6341     0.3639     0.8729         16        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.856      0.646      0.717      0.476

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      7.41G     0.6079     0.3561     0.8628         24        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.44it/s]


                   all       1274       7377      0.878      0.636      0.727      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      7.28G     0.6055     0.3434     0.8669         17        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.865       0.64      0.721      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      7.45G     0.6064     0.3446     0.8638         12        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.876       0.64      0.722      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      7.44G     0.6003     0.3437     0.8682         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.48it/s]


                   all       1274       7377      0.882      0.631       0.72      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      7.28G     0.5926     0.3421     0.8689         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.877      0.641      0.722      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      7.28G     0.6013     0.3433     0.8661          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377       0.88       0.64      0.723      0.484

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      7.27G     0.6067     0.3441     0.8563          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.876      0.643      0.723      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      7.24G        inf     0.3457     0.8598         22        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.45it/s]


                   all       1274       7377       0.87      0.638      0.717      0.479

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      7.36G     0.6131      0.341     0.8804          4        640: 100%|██████████| 36/36 [00:08<00:00,  4.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.886      0.636      0.725      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      7.43G     0.6038     0.3414     0.8668         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.65it/s]


                   all       1274       7377      0.886      0.628      0.725      0.484

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      7.27G     0.5878     0.3361     0.8618         36        640: 100%|██████████| 36/36 [00:08<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377      0.878      0.644      0.727      0.484

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      7.26G     0.5904     0.3337     0.8633          7        640: 100%|██████████| 36/36 [00:08<00:00,  4.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.888      0.637      0.732      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      7.28G     0.5894     0.3347     0.8596          9        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377        0.9      0.635      0.732      0.488

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      7.28G     0.5865     0.3303     0.8597          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.60it/s]


                   all       1274       7377      0.885      0.637      0.731      0.488

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      7.44G     0.5869     0.3394     0.8604          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.46it/s]


                   all       1274       7377      0.895      0.635      0.735       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      7.39G     0.5888     0.3293     0.8659         11        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.60it/s]


                   all       1274       7377      0.868      0.644       0.73      0.491

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      7.26G     0.6335      0.383     0.9259          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.49it/s]


                   all       1274       7377      0.878      0.642      0.726      0.487

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      7.36G     0.5861     0.3317     0.8618         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.49it/s]


                   all       1274       7377      0.874      0.642      0.731      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      7.26G     0.5848     0.3258     0.8594         16        640: 100%|██████████| 36/36 [00:08<00:00,  4.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.867      0.652      0.733       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      7.44G     0.5531     0.3106     0.8608         14        640: 100%|██████████| 36/36 [00:08<00:00,  4.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.874      0.648      0.737      0.491

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      7.25G     0.5584     0.3221     0.8659          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377      0.873      0.648      0.734      0.492
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      7.26G     0.6291     0.3287     0.8444         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.49it/s]


                   all       1274       7377      0.889      0.642      0.732      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100      7.35G     0.6304     0.3264     0.8498         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.51it/s]


                   all       1274       7377       0.89       0.64      0.732       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      7.26G     0.6268     0.3258     0.8426          7        640: 100%|██████████| 36/36 [00:07<00:00,  4.54it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.52it/s]


                   all       1274       7377       0.87      0.647      0.736      0.492

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      7.28G     0.6328     0.3278     0.8444          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.60it/s]


                   all       1274       7377      0.855      0.658      0.736      0.493

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100       7.3G     0.6327     0.3278      0.844         31        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.01it/s]


                   all       1274       7377       0.87      0.644      0.734      0.492

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      7.41G     0.6183     0.3317     0.8465         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.42it/s]


                   all       1274       7377      0.863      0.655      0.738      0.495

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      7.41G     0.6261     0.3262     0.8435         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377       0.87      0.652       0.74      0.496

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      7.41G     0.6205     0.3218     0.8365          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.58it/s]


                   all       1274       7377      0.862      0.656       0.74      0.495

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      7.36G      0.615     0.3152     0.8408         40        640: 100%|██████████| 36/36 [00:08<00:00,  4.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.19it/s]


                   all       1274       7377      0.862      0.656      0.741      0.494

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      7.35G     0.6089     0.3105     0.8451          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.90it/s]


                   all       1274       7377      0.867      0.655       0.74      0.495

100 epochs completed in 0.472 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/last.pt, 22.5MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt, 22.5MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  2.89it/s]


                   all       1274       7377       0.87      0.653      0.739      0.496
            Pedestrian        333        600      0.814      0.688      0.751      0.445
         Traffic_Signs        309        434      0.847      0.585      0.677      0.408
               Vehicle       1034       2820      0.929      0.882      0.933      0.763
         traffic_light        660       3523      0.888      0.455      0.596      0.367
Speed: 0.1ms preprocess, 1.2ms inference, 0.0ms loss, 1.7ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5


# Results on Validation set:
100 epochs completed in 0.472 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/last.pt, 22.5MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt, 22.5MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:06<00:00,  2.89it/s]
                   all       1274       7377       0.87      0.653      0.739      0.496
            Pedestrian        333        600      0.814      0.688      0.751      0.445
         Traffic_Signs        309        434      0.847      0.585      0.677      0.408
               Vehicle       1034       2820      0.929      0.882      0.933      0.763
         traffic_light        660       3523      0.888      0.455      0.596      0.367
Speed: 0.1ms preprocess, 1.2ms inference, 0.0ms loss, 1.7ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [12]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<00:00, 931.54it/s]


val: New cache created: /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.67s/it]


                   all        641       3701      0.868       0.63      0.776      0.565
            Pedestrian        182        332      0.784       0.63      0.752      0.504
         Traffic_Signs        168        230      0.864      0.578      0.748      0.494
               Vehicle        528       1377      0.944      0.872       0.93      0.805
         traffic_light        338       1762      0.879      0.439      0.675      0.457
Speed: 0.3ms preprocess, 2.2ms inference, 0.0ms loss, 1.8ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<00:00, 931.54it/s]
val: New cache created: /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.67s/it]
                   all        641       3701      0.868       0.63      0.776      0.565
            Pedestrian        182        332      0.784       0.63      0.752      0.504
         Traffic_Signs        168        230      0.864      0.578      0.748      0.494
               Vehicle        528       1377      0.944      0.872       0.93      0.805
         traffic_light        338       1762      0.879      0.439      0.675      0.457
Speed: 0.3ms preprocess, 2.2ms inference, 0.0ms loss, 1.8ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [13]:
from ultralytics import YOLO

# Load the model
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt")

# Access the epoch information
checkpoint = model.ckpt  # This loads the checkpoint file as a dictionary
best_epoch = checkpoint['epoch'] if 'epoch' in checkpoint else "Epoch info not found"
print(f"Best checkpoint was saved at epoch: {best_epoch}")


Best checkpoint was saved at epoch: -1


In [14]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.60s/it]


                   all        641       3701      0.855      0.634      0.776      0.564
            Pedestrian        182        332      0.747      0.642       0.75      0.505
         Traffic_Signs        168        230      0.853      0.578      0.749      0.491
               Vehicle        528       1377      0.944      0.874      0.931      0.804
         traffic_light        338       1762      0.877      0.442      0.675      0.456
Speed: 0.3ms preprocess, 2.0ms inference, 0.0ms loss, 1.8ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv8s - 0.241
- IoU threshold : 0.5
- Results save d at: runs/detect/val3

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.241, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:06<00:00,  1.16s/it]


                   all        641       3701      0.861      0.632      0.777      0.565
            Pedestrian        182        332      0.773      0.635      0.755      0.505
         Traffic_Signs        168        230       0.85      0.578      0.747      0.494
               Vehicle        528       1377      0.944      0.874      0.931      0.806
         traffic_light        338       1762      0.877      0.443      0.675      0.457
Speed: 2.0ms preprocess, 2.4ms inference, 0.0ms loss, 2.3ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best + Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv8s - 0.241
- IoU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.241, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]


                   all        641       3701      0.855      0.635      0.777      0.564
            Pedestrian        182        332      0.747      0.642      0.751      0.505
         Traffic_Signs        168        230      0.848      0.583       0.75      0.491
               Vehicle        528       1377      0.948      0.874      0.931      0.804
         traffic_light        338       1762      0.877      0.444      0.676      0.456
Speed: 2.0ms preprocess, 2.2ms inference, 0.0ms loss, 1.8ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best Epoch 97 +  Optimal Confidence + IoU 0.6
- Optimal Confidence YOLOv8s - 0.241
- IoU threshold : 0.6
- Results save d at: runs/detect/val5

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.241, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.24s/it]


                   all        641       3701       0.85      0.636      0.776      0.563
            Pedestrian        182        332      0.732      0.642      0.748      0.503
         Traffic_Signs        168        230      0.848      0.583       0.75      0.491
               Vehicle        528       1377      0.943      0.875      0.931      0.804
         traffic_light        338       1762      0.876      0.444      0.676      0.456
Speed: 1.9ms preprocess, 2.5ms inference, 0.0ms loss, 1.8ms postprocess per image
Saving runs/detect/val5/predictions.json...
Results saved to runs/detect/val5


# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_v8/v8_s/train5/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 11,127,132 parameters, 0 gradients, 28.4 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:28<00:00,  1.26it/s]


                   all       4485      26788      0.907      0.678      0.817      0.629
            Pedestrian       1229       2227      0.857      0.741      0.843      0.616
         Traffic_Signs       1093       1459      0.897      0.622      0.785      0.559
               Vehicle       3636      10031       0.96      0.905      0.949      0.842
         traffic_light       2373      13071      0.915      0.442      0.691      0.501
Speed: 0.4ms preprocess, 2.1ms inference, 0.0ms loss, 0.7ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.